onnxruntime/core/providers/rocm/math/matmul.cc

```diff
@@ -2,9 +2,8 @@
 // Licensed under the MIT License.
 
 #include "core/providers/rocm/math/matmul.h"
-#include "core/providers/cpu/math/matmul_helper.h"
```

Comment:
helper is used in this file, so keep the header file?

Reply:
It's moved to "matmul_impl.h", and this file include "matmul_impl.h", so I think we don't need it there.

Reply:
I think it's still good to include it according to this style guide:

Do not rely on transitive inclusions. This allows people to remove no-longer-needed #include statements from their headers without breaking clients. This also applies to related headers - foo.cc should include bar.h if it uses a symbol from it even if foo.h includes bar.h.

https://google.github.io/styleguide/cppguide.html#Include_What_You_Use

Reply:
@zhangyaobit Actually, we, onnxruntime, do not have an established discipline for the header files, set aside IWYU. ORT heavily rely on forward declaration, and most of the headers are not self contained, XDDDDDD. And it is generally a pain in the ass to figure out the inclusion order, again XDDDD. We need someone to step foward to clean those things up. Better rely on established tools and integrate it in the CI, instead of manaully checking it here. 

Anyway, IWYU is a pretty good practice. And should be enforced at the level of ORT at a whole.